instrumentation/netty/netty-4.1/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/netty/v4_1/Netty41ConnectionSpanTest.java

```diff
@@ -0,0 +1,237 @@
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+class Netty41ConnectionSpanTest {
+  @RegisterExtension
+  static final InstrumentationExtension testing = AgentInstrumentationExtension.create();
+
+  private HttpClientTestServer server;
+  private EventLoopGroup eventLoopGroup;
+
+  private Bootstrap bootstrap;
+
+  @BeforeEach
+  void setUp() {
+    eventLoopGroup = new NioEventLoopGroup();
+    bootstrap = buildBootstrap();
+    server = new HttpClientTestServer(testing.getOpenTelemetry());
+    server.start();
+  }
+
+  @AfterEach
+  void tearDown() {
+    eventLoopGroup.shutdownGracefully();
+    server.stop();
+  }
+
+  Bootstrap buildBootstrap() {
+    Bootstrap bootstrap = new Bootstrap();
+    bootstrap
+        .group(eventLoopGroup)
+        .channel(NioSocketChannel.class)
+        .handler(
+            new ChannelInitializer<SocketChannel>() {
+              @Override
+              protected void initChannel(SocketChannel socketChannel) throws Exception {
+                ChannelPipeline pipeline = socketChannel.pipeline();
+                pipeline.addLast(new HttpClientCodec());
+              }
+            });
+
+    return bootstrap;
+  }
+
+  DefaultFullHttpRequest buildRequest(String method, URI uri, Map<String, String> headers) {
+    DefaultFullHttpRequest request =
+        new DefaultFullHttpRequest(
+            HttpVersion.HTTP_1_1, HttpMethod.valueOf(method), uri.getPath(), Unpooled.EMPTY_BUFFER);
+    request.headers().set(HttpHeaderNames.HOST, uri.getHost() + ":" + uri.getPort());
+    headers.forEach((k, v) -> request.headers().set(k, v));
+    return request;
+  }
+
+  int sendRequest(DefaultFullHttpRequest request, URI uri)
+      throws ExecutionException, InterruptedException, TimeoutException {
+    Channel channel = bootstrap.connect(uri.getHost(), uri.getPort()).sync().channel();
+    CompletableFuture<Integer> result = new CompletableFuture<>();
+    channel.pipeline().addLast(new ClientHandler(result));
+    channel.writeAndFlush(request).get();
+    return result.get(20, TimeUnit.SECONDS);
+  }
+
+  @Test
+  @DisplayName("test successful request")
+  void testSuccessfulRequest() throws Exception {
+    URI uri = URI.create("http://localhost:" + server.httpPort() + "/success");
+    DefaultFullHttpRequest request = buildRequest("GET", uri, new HashMap<>());
+    int responseCode = testing.runWithSpan("parent", () -> sendRequest(request, uri));
+
+    assertThat(responseCode).isEqualTo(200);
+    testing.waitAndAssertTraces(
+        trace ->
+            trace
+                .hasSize(5)
```

Comment:
`hasSize` is unnecessary because `hasSpansSatisfyingExactly` will check the size. It seem there are many same problems, please modify them together.